File: backend/StockManagementSystem/src/main/java/Assignment/StockManagementSystem/services/InventoriesService.java
package Assignment.StockManagementSystem.services;

import Assignment.StockManagementSystem.dto.InventoryBulkUpdateDTO;
import Assignment.StockManagementSystem.dto.InventoryDTOWithoutId;
import Assignment.StockManagementSystem.dto.InventoryDTOWithoutItems;
import Assignment.StockManagementSystem.models.Inventories;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
public interface InventoriesService {

    Inventories addInventory(InventoryDTOWithoutId inventoryRequestDTO);

    Page<InventoryDTOWithoutItems> getInventories(String searchTerm, Pageable pageable);

    Inventories updateInventory(int inventoryId, InventoryBulkUpdateDTO updatedInventory);

}

File: frontend/stock_management_system_frontend/src/Services/InventoriesService.js
import api from "../APIs/AxiosInstance";

export const fetchInventories = async ({ searchTerm, pagination }) => {
  const params = {
    searchTerm: searchTerm,
    page: pagination?.page || 0,
    size: pagination?.size || 10,
  };

  const response = await api.get("/inventories", { params });
  return response.data;
};

export const addInventory = async (inventoryData) => {
  const response = await api.post("/inventories", inventoryData);
  return response.data;
};

export const updateInventory = async (inventoryId, updatedData) => {
  const response = await api.put(`/inventories/${inventoryId}`, updatedData);
  return response.data;
};
